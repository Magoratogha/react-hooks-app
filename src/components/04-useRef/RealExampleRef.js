import { Fragment, useState } from "react";
import MultipleCustomHooks from "../03-examples/MultipleCustomHooks";
import './FocusScreen.css';

function RealExampleRef() {

    const [show, setShow] = useState(false);

    return (
        <Fragment>
            <h1>RealExampleRef</h1>
            <hr></hr>
            { show && <MultipleCustomHooks /> 
            }
            <div className="row">
                <div className="col-12">
                    <button 
                        onClick={ () => setShow(!show) }
                        className="btn btn-outline-dark mt-3">
                            Show/Hide
                    </button>
                </div>
            </div>
        </Fragment>
    )
}

export default RealExampleRef;