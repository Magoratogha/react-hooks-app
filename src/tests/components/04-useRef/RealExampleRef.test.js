import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import RealExampleRef from "../../../components/04-useRef/RealExampleRef";

describe('Test on <RealExampleRef />', () => {

    const user = userEvent.setup();

    test('should render properly', () => {
        const { container } = render(<RealExampleRef />);
        expect(container).toMatchSnapshot();
    });

    test('should render <MultipleCustomHooks />', async () => {
        render(<RealExampleRef />);
        const buttonElement = screen.queryByText('Show/Hide', { selector: 'button' });
        await user.click(buttonElement);
        expect(screen.getByText('Breaking Bad quotes', {selector: 'h1'})).toBeTruthy();
    });
});