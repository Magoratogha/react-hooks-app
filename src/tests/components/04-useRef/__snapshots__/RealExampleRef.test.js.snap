// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on <RealExampleRef /> should render properly 1`] = `
<div>
  <h1>
    RealExampleRef
  </h1>
  <hr />
  <div
    class="row"
  >
    <div
      class="col-12"
    >
      <button
        class="btn btn-outline-dark mt-3"
      >
        Show/Hide
      </button>
    </div>
  </div>
</div>
`;
